integration-tests/src/dynamic_fees.rs

```diff
@@ -0,0 +1,424 @@
+#![cfg(test)]
+
+use crate::{oracle::hydradx_run_to_block, polkadot_test_net::*};
+use frame_support::assert_ok;
+use pallet_dynamic_fees::types::FeeEntry;
+use primitives::AssetId;
+use sp_runtime::{FixedU128, Permill};
+use xcm_emulator::TestExt;
+
+const DOT_UNITS: u128 = 10_000_000_000;
+const BTC_UNITS: u128 = 10_000_000;
+const ETH_UNITS: u128 = 1_000_000_000_000_000_000;
+
+
+#[test]
+fn fees_should_work_when_oracle_not_initialized() {
+	TestNet::reset();
+
+	Hydra::execute_with(|| {
+		//Arrange
+		init_omnipool();
+
+		let trader = DAVE;
+
+		set_balance(trader.into(), DOT, 1_000 * DOT_UNITS as i128);
+
+		assert!(hydradx_runtime::DynamicFees::current_fees(HDX).is_none());
+
+		//Act
+		assert_ok!(hydradx_runtime::Omnipool::sell(
+			hydradx_runtime::Origin::signed(DAVE.into()),
+			DOT,
+			HDX,
+			2 * DOT_UNITS,
+			0,
+		));
+
+		// REVIEW: IS THIS CORRECT? 
+		// If yes, are you testing this correctly in other tests? 
+		// looks like you are not testing the change or maybe this is a special case when not initialized?
+		assert!(hydradx_runtime::DynamicFees::current_fees(HDX).is_none());
```

Comment:
Just this, other than that LGTM

Reply:
this is special case. it could happen on first trade when there are no data in oracle yet. 

in this case, min fee should be applied. this case is tested within dynamic fees pallet, where it ensures that min fee is returned if no data available.

in production- this wont happen there are already oracle data.

Reply:
oh yes. the min fee will be applied in that case.

I will add additional test, more obvious one.